src/osd/PG.cc

```diff
@@ -7697,6 +7697,12 @@ boost::statechart::result
 PG::RecoveryState::Recovering::react(const UnfoundRecovery &evt)
 {
   PG *pg = context< RecoveryMachine >().pg;
+  if (!pg->state_test(PG_STATE_RECOVERING)) {
+    // we may have finished recovery and have an AllReplicasRecovered
+    // event queued to move us to the next state.
+    ldout(pg->cct, 10) << "got defer recovery but not recovering" << dendl;
```

Comment:
s/defer/unfound/, or did you mean to add this to other states too?

Reply:
whoops, just added it to the wrong state reaction.